### PR TITLE
[[ Community Docs ]] repeat - various fixes

### PR DESCRIPTION
**Pull request by asayd**
- Revised note about changing a container variable inside a `for each` loop.
- Added change note re the previous.
- Major adjustments to formatting to improve readability.
- Beefed up references and links.
- Added examples, modified examples to bring into compliance with guidelines.
- Minor wording changes for clarity.

### CODE REVIEW

**Comment by asayd** [comment]
@runrevmark - I added a note and an entry to the Changes element to reflect the change in v. 7 regarding container variables in `for each` loops. Does this capture the change accurately? You mentioned it in the use-livecode 'Changing local variables' thread.

**Comment by asayd** [comment]
@livecodeali - 
The formatting in this entry was pretty messed up, even as compared to the pre-8 version of the dictionary. I've tried to clean it up, but I wonder if I am overdoing it. For example, using code tags in the Parameter section, and indeed throughout the Description narrative. I like how it can differentiate between actual scripting terms, variable data, and other text, but is it overkill? Does it start to look too blocky? Would it be just as good to leave it in plain text and italicize the variables?

This actually brings up the larger question of style and the need for style guidelines. For example, if a scripting term is encountered in the description, is it best to mark it as `code` or to add that term to the References and link the term? Does such a style guide exist? I've been studying the Format Reference document of course, but there are plenty of subtleties that aren't covered there.

**Comment by asayd** [comment]
@livecodeali I just had a reread of the Format Reference guide, and found that I had overlooked this sentence: "The Description element is the only one that allows standard markdown." In that case some of the changes in this PR violate that rule. But the problem in entries like this one that have complex lists of parameters is that, as it stands now, the parameter descriptions often have gotten munged into one long, hard-to-read paragraph. So I used some list and code markdown to try to sort them out better. I hope this is okay; some of the entries really need some tidying up, specifically in Parameters and Values. 

Forgive all the questions. I want to make sure I'm getting these right, and hope I can help any other contributors with style issues.

**Comment by livecodeali** [comment]
@asayd Hmm, yes the document format reference does need an update in that respect. Perhaps unfortunately, there is special parsing/formatting in place for parameters of `enum` and `array` type (although the latter doesn't currently work). This overrides the markdown, so you can put markdown in the description of a parameter provided it is neither of these types.

Currently `Changes:` cannot contain markdown, although I think it ought to be able to. 

In terms of marking things as `<code>` or not, I'm not quite sure. I'd say do as you have done here with everything that really is code - we can always change the display of inline code blocks so that they're less obtrusive (I have to say, I really don't like the red-on-pink that we have at the moment- I think that's a Bootstrap thing).

So in summary, I think the only change really necessary here is changing `loopForm (enum):` to just `loopForm:` in the params list.

File: docs/dictionary/control_st/repeat.lcdoc [hunk 1]
```diff
@@ -25,124 +55,222 @@ repeat with tVarNum = 1 to 20
...
+####Loop Form:
+The <repeat> <control structure> always begins with the `repeat`
+<keyword>.
+The last line of a <repeat> <control structure> is always the `end
+repeat` <keyword>.
+
+How many times the <statementList> is <execute|executed> depends on the
+<loopForm> you use. Here are the possible forms:
+
+#####<Forever>
```
Comment (livecodeali) [inline docs/dictionary/control_st/repeat.lcdoc]: This will need to be `<forever|Forever>` to link correctly

File: docs/dictionary/control_st/repeat.lcdoc [hunk 2]
```diff
@@ -25,124 +55,222 @@ repeat with tVarNum = 1 to 20
...
+`the mouseClick` is not checked until after the `go` command:
+
+    repeat forever
+      go next card
+      if the mouseClick then exit repeat -- user clicked
+    end repeat
+
+If no <loopForm> is specified, the `forever` form is used.
+
+#####<Until> and <while>
```
Comment (livecodeali) [inline docs/dictionary/control_st/repeat.lcdoc]: likewise, `<until|Until>`

File: docs/dictionary/control_st/repeat.lcdoc [hunk 3]
```diff
@@ -25,124 +55,222 @@ repeat with tVarNum = 1 to 20
...
-- "for each element labelVariable in array"
-- "for each key labelVariable in array"
-statementList: One or more LiveCode statements, and can also include if, switch, try, or repeat control structures.
-condition (bool): 
-number, startValue, endValue, and increment: Numbers or expressions that evaluate to numbers.
-counter or labelVariable: A legal variable name.
-chunkType: One of byte, codeunit, codepoint, character (or char), token, trueword, word (or segment), item, sentence, paragraph or line.
-container: Any existing container.
-array (array): Any existing container that contains an array of values.
+loopForm (enum):
```
Comment (livecodeali) [inline docs/dictionary/control_st/repeat.lcdoc]: Sorry, could you remove this `(enum)` too please? That should make the markdown render correctly.

**Comment by livecodeali** [comment]
@asayd Hi Devin, could you remove the `(enum)` from `loopForm (enum)` please- then we can get this reviewed and merged. Thanks!

**Comment by livecodeali** [comment]
@livecode-vulcan review ok f9ee57b

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok f9ee57b1bf4666412346ee5e93206e04b2375dc0

**Comment by livecode-vulcan** [comment]
😎 test success f9ee57b1bf4666412346ee5e93206e04b2375dc0
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2474)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2484)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2467)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2485)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2491)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2495)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2507)
